src/TStringUtils.h

```diff
@@ -54,6 +54,29 @@ class TStringUtils
             func(ptr);
         }
     }
+
+    static void listAddItem(QString &list, const QString &item)
+    {
+        if (!list.isEmpty()) {
+            list.append("|");
+        }
+        list.append(item);
+    }
+
+    static void listRemoveItem(QString &list, const QString &item)
+    {
+        int start = list.indexOf(item);
+
+        if (start != -1) {
+            if (start == 0) { // in the start of string
+                if (item.size() == list.size() || list[item.size()] == '|') {
+                    list.remove(start, std::min(item.size() + 1, list.size()));
+                }
+            } else if (list[start - 1] == '|') { // in the middle/end of string
+                list.remove(start - 1, item.size() + 1);
+            }
+        }
```

Comment:
Avoid using the non-`const` accessor `operator[int]` when you don't need to and use the faster `const` `at(int)` instead - and by returning early when the item is not found you can simplify the code by removing one level of braces in what is left:
```suggestion
        if (start == -1) {
            return;
        }
        if (!start) { // in the start of string
            if (item.size() == list.size() || list.at(item.size()) == '|') {
                list.remove(start, std::min(item.size() + 1, list.size()));
            }
        } else if (list.at(start - 1) == '|') { // in the middle/end of string
            list.remove(start - 1, item.size() + 1);
        }
```